modules/motions/MiniFigCameraAction.cpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * @file   MiniFigCameraAction.cpp
+ * @brief  配置エリアAにおけるミニフィグ撮影動作
+ * @author nishijima515
+ */
+
+#include "MiniFigCameraAction.h"
+
+using namespace std;
+
+MiniFigCameraAction::MiniFigCameraAction(Robot& _robot, bool _isClockwise, int _preTargetAngle,
+                                         int _postTargetAngle, double _targetRotationSpeed,
+                                         double _backTargetDistance, double _forwardTargetDistance,
+                                         double _backSpeed, double _forwardSpeed, int position)
+  : CompositeMotion(_robot),
+    isClockwise(_isClockwise),
+    preTargetAngle(_preTargetAngle),
+    postTargetAngle(_postTargetAngle),
+    targetRotationSpeed(_targetRotationSpeed),
+    backTargetDistance(_backTargetDistance),
+    forwardTargetDistance(_forwardTargetDistance),
+    backSpeed(_backSpeed),
+    forwardSpeed(_forwardSpeed),
+    position(position)
+{
+}
+
+bool MiniFigCameraAction::isMetPreCondition()
+{
+  if(position != 0 && robot.getMiniFigDirectionResult().wasDetected
+     && robot.getMiniFigDirectionResult().direction != static_cast<MiniFigDirection>(position)) {
```

Comment:
wasDetected は false で初期化しているから、position == 0 の時って、確定で wasDetected == false になるよね

```suggestion
  if(robot.getMiniFigDirectionResult().wasDetected
     && robot.getMiniFigDirectionResult().direction != static_cast<MiniFigDirection>(position)) {
```